Sources/SnapshotTesting/Strategy.swift

```diff
@@ -1,59 +1,90 @@
 import Foundation
 import XCTest
 
+/// A type representing the ability to transform a value into a diffable format (like text or an image) for snapshot testing.
 public struct Strategy<Snapshottable, Format> {
+  /// The path extension applied to references saved to disk.
   public var pathExtension: String?
+
+  /// How the snapshot format is diffed and converted to and from data.
   public let diffable: Diffable<Format>
-  public let snapshotToDiffable: (Snapshottable) -> Async<Format>
 
+  /// How a value is transformed into a diffable snapshot format.
+  public let snapshot: (Snapshottable) -> Async<Format>
+
+  /// Creates a snapshot strategy.
+  ///
+  /// - Parameters:
+  ///   - pathExtension: The path extension applied to references saved to disk.
+  ///   - diffable: How to diff and convert the snapshot format to and from data.
+  ///   - snapshot: An asynchronous transform function from a value into a diffable snapshot format.
+  ///   - value: A value to be converted.
   public init(
     pathExtension: String?,
     diffable: Diffable<Format>,
-    snapshotToDiffable: @escaping (Snapshottable) -> Async<Format>
+    snapshot: @escaping (_ value: Snapshottable) -> Async<Format>
     ) {
     self.pathExtension = pathExtension
     self.diffable = diffable
-    self.snapshotToDiffable = snapshotToDiffable
+    self.snapshot = snapshot
   }
 
+  /// Creates a snapshot strategy.
+  ///
+  /// - Parameters:
+  ///   - pathExtension: The path extension applied to references saved to disk.
+  ///   - diffable: How to diff and convert the snapshot format to and from data.
+  ///   - snapshot: A transform function from a value into a diffable snapshot format.
+  ///   - value: A snapshot value to be converted.
   public init(
     pathExtension: String?,
     diffable: Diffable<Format>,
-    snapshotToDiffable: @escaping (Snapshottable) -> Format
+    snapshot: @escaping (_ value: Snapshottable) -> Format
     ) {
     self.init(pathExtension: pathExtension, diffable: diffable) {
-      Async(value: snapshotToDiffable($0))
+      Async(value: snapshot($0))
     }
   }
 
-  public func asyncPullback<A>(_ transform: @escaping (A) -> Async<Snapshottable>) -> Strategy<A, Format> {
+  /// Transforms a strategy on `Value`s into a strategy on `A`s through a function `(A) -> Async<Value>`.
```

Comment:
It's more like:

```suggestion
  /// Transforms a strategy on `Snapshottable`s into a strategy on `A`s through a function `(A) -> Async< Snapshottable>`.
```

right? do you think people will be confused with the differing type names?

Reply:
Ah yeah. Should we rethink the names?

Reply:
I guess `Strategy<Value, Format>` still reads really nicely.

Can't remember if we had a bigger reason for going with the super verbose `Snapshottable`. it's also not great that it ends in `-able`, makes it sound like a protocol.

Reply:
I like `Value`! Can we also fix the naming on `Diffable`?

Reply:
`Diffing` would prob do. `Diffing<String>` and `Diffing<UIImage>` reads well.

Reply:
even `Snapshotting` is kinda nice instead of `Strategy` lol. but don't know if we want to change that.